### PR TITLE
Problem: generic events format are not implemented

### PR DESCRIPTION
**Pull request by thomas-nguy**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

Solution : Implement ADR-007

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: contracts/src/ModuleCRC21.sol [hunk 1]
```diff
@@ -8,9 +8,9 @@ contract ModuleCRC21 is DSToken {
     string denom;
     bool isSource;
 
-    event __CronosSendToIbc(address sender, string recipient, uint256 amount);
-    event __CronosSendToChain(address sender, address recipient, uint256 amount, uint256 bridge_fee, uint256 chain_id);
-    event __CronosCancelSendToChain(address sender, uint256 id);
+    event __CronosSendToIbc(address sender, string recipient, uint256 amount, string channel_id, bytes extraData);
```
Comment (thomas-nguy) [inline contracts/src/ModuleCRC21.sol:11]: Will support indexed params in another PR. This is quite a big change.

File: contracts/src/ModuleCRC21.sol [hunk 2]
```diff
@@ -8,9 +8,9 @@ contract ModuleCRC21 is DSToken {
...
+    event __CronosSendToEvmChain(address sender, address recipient, uint256 chain_id, uint256 amount, uint256 bridge_fee, bytes extraData);
```
Comment (JayT106) [inline contracts/src/ModuleCRC21.sol:12]: missing `indexed` in these 2 events?

File: x/cronos/keeper/evmhandlers/send_to_ibc_v2.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	bytesType, _ := abi.NewType("bytes", "", nil)
+
+	SendToIbcEventV2 = abi.NewEvent(
+		SendToIbcEventName,
+		SendToIbcEventName,
+		false,
+		abi.Arguments{abi.Argument{
+			Name:    "sender",
+			Type:    addressType,
+			Indexed: false,
```
Comment (JayT106) [inline x/cronos/keeper/evmhandlers/send_to_ibc_v2.go:35]: set the `Indexed` to `true` and the following arguments?

File: x/cronos/keeper/evmhandlers/send_to_ibc_v2.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	if !types.IsValidIBCDenom(denom) && !types.IsValidCronosDenom(denom) {
+		return fmt.Errorf("the native token associated with the contract %s is neither an ibc voucher or a cronos token", contract)
+	}
+
+	contractAddr := sdk.AccAddress(contract.Bytes())
+	sender := sdk.AccAddress(unpacked[0].(common.Address).Bytes())
+	recipient := unpacked[1].(string)
+	amount := sdk.NewIntFromBigInt(unpacked[2].(*big.Int))
+	// channelId := unpacked[3].(string)
+	// extraData := unpacked[4].([]byte)
```
Comment (JayT106) [inline x/cronos/keeper/evmhandlers/send_to_ibc_v2.go:100]: why didn't we handle these 2 data?

Comment (thomas-nguy) [reply x/cronos/keeper/evmhandlers/send_to_ibc_v2.go:100]: arguments check is already handled by this line 
```
unpacked, err := SendToIbcEventV2.Inputs.Unpack(data)
```

this is a comment to help for future use

**Review by adu-crypto — APPROVED** [review]
lgtm

File: x/cronos/keeper/evmhandlers/send_to_ibc_v2.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (thomas-nguy) [reply x/cronos/keeper/evmhandlers/send_to_ibc_v2.go:35]: yes, indexed events are going to be implemented in the next PR (still in progress)

**Pull request closed without merging**